### PR TITLE
Create typing reducer and keep flag on inserting blocks

### PR DESCRIPTION
**Pull request by ellatrix**
See #1561.

To test: type in a text block and press enter twice to create a new text block. No UI should appear.

### CODE REVIEW

**Review by youknowriad — COMMENTED** [review]
This works pretty well. Two things I think we should do (maybe later)

- Handle Tab and Arrow keys: The behaviour is not inconsistent between the types of blocks, if we navigate to a text block, the borders stay hidden but if we navigate to an image block or a "blocky" block they appear again (and stay even if we navigate to a text block again)
- Maybe hide the placeholder of the text block if we're typing

**Comment by iseulde** [comment]
@youknowriad It seems like scroll is triggering it sometimes? Not sure why, only mouse move should trigger it. Maybe we should add a buffer at some point to only trigger mouse move after # px.

File: editor/selectors.js [hunk 1]
```diff
@@ -544,12 +544,8 @@ export function getBlockFocus( state, uid ) {
  * @param  {Object}  uid   Block unique ID
  * @return {Boolean}       Whether user is typing within block
  */
-export function isTypingInBlock( state, uid ) {
-	if ( ! isBlockSelected( state, uid ) ) {
-		return false;
-	}
-
-	return state.selectedBlock.typing;
+export function isTypingInEditor( state ) {
```
Comment (aduth) [inline editor/selectors.js]: Could we name this simply `isTyping`? "In editor" seems implied.

Comment (iseulde) [inline editor/selectors.js]: Prop name suggestions for components using it?

Comment (aduth) [inline editor/selectors.js]: Also, DocBlock should be updated.

Comment (aduth) [inline editor/selectors.js]: `isTyping` is fine if we don't insist to destructure props.

Comment (iseulde) [inline editor/selectors.js]: Yeah they are being destructed. So either we rename it there, or change it to non destructing.

**Comment by iseulde** [comment]
Anything left here?

**Review by aduth — APPROVED** [review]
Code-wise looks great; nice simplification.

Seems to be a good direction for writing flow too. I agree with @youknowriad about placeholder: seems odd to appear with this flow now.

**Comment by mtias** [comment]
My only concern is that the setup with single line breaks feels more weird since it's not evident when you are in a new block / new paragraph. But let's try it.

What if we rename the placeholder to "New paragraph"?

**Comment by iseulde** [comment]
@mtias I would keep the placeholders in general for consistency on backspace and focus issues with buttons if we take the focus approach. I don't mind "Write..." for now or maybe renaming as your proposing. Either placeholder text is fine for me.

**Comment by mtias** [comment]
This is slightly clear to me:

image

But ok with merging either way and testing how it feels. (It will expose the line-break / paragraph break differences.)

**Pull request merged**